engine/src/main/java/org/pentaho/di/job/entries/folderscompare/JobEntryFoldersCompare.java

```
@@ -0,0 +1,628 @@
+    } catch ( KettleXMLException xe ) {
+      throw new KettleXMLException( BaseMessages.getString( PKG, "JobFoldersCompare.Meta.UnableLoadXML", xe
+        .getMessage() ) );
+    }
+  }
+
+  public void loadRep( Repository rep, IMetaStore metaStore, ObjectId id_jobentry, List<DatabaseMeta> databases,
+    List<SlaveServer> slaveServers ) throws KettleException {
+    try {
+      includesubfolders = rep.getJobEntryAttributeBoolean( id_jobentry, "include_subfolders" );
+      comparefilecontent = rep.getJobEntryAttributeBoolean( id_jobentry, "compare_filecontent" );
+      comparefilesize = rep.getJobEntryAttributeBoolean( id_jobentry, "compare_filesize" );
+
+      compareonly = rep.getJobEntryAttributeString( id_jobentry, "compareonly" );
+      wildcard = rep.getJobEntryAttributeString( id_jobentry, "wildcard" );
+      filename1 = rep.getJobEntryAttributeString( id_jobentry, "filename1" );
+      filename2 = rep.getJobEntryAttributeString( id_jobentry, "filename2" );
+    } catch ( KettleException dbe ) {
+      throw new KettleException( BaseMessages.getString( PKG, "JobFoldersCompare.Meta.UnableLoadRep", ""
+        + id_jobentry, dbe.getMessage() ) );
+    }
+  }
+
+  public void saveRep( Repository rep, IMetaStore metaStore, ObjectId id_job ) throws KettleException {
+    try {
+      rep.saveJobEntryAttribute( id_job, getObjectId(), "include_subfolders", includesubfolders );
+      rep.saveJobEntryAttribute( id_job, getObjectId(), "compare_filecontent", comparefilecontent );
+      rep.saveJobEntryAttribute( id_job, getObjectId(), "compare_filesize", comparefilesize );
+
+      rep.saveJobEntryAttribute( id_job, getObjectId(), "compareonly", compareonly );
+      rep.saveJobEntryAttribute( id_job, getObjectId(), "wildcard", wildcard );
+      rep.saveJobEntryAttribute( id_job, getObjectId(), "filename1", filename1 );
+      rep.saveJobEntryAttribute( id_job, getObjectId(), "filename2", filename2 );
+    } catch ( KettleDatabaseException dbe ) {
```

Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)